internal/resources/metal/project_ssh_key/models.go

```diff
@@ -19,14 +19,14 @@ type ResourceModel struct {
 	ProjectID   types.String `tfsdk:"project_id"`
 }
 
-func (m *ResourceModel) parse(key *packngo.SSHKey) diag.Diagnostics {
-	m.ID = types.StringValue(key.ID)
-	m.Name = types.StringValue(key.Label)
-	m.PublicKey = types.StringValue(key.Key)
-	m.Fingerprint = types.StringValue(key.FingerPrint)
-	m.Created = types.StringValue(key.Created)
-	m.Updated = types.StringValue(key.Updated)
-	m.OwnerID = types.StringValue(path.Base(key.Owner.Href))
+func (m *ResourceModel) parse(key *metalv1.SSHKey) diag.Diagnostics {
+	m.ID = types.StringValue(key.GetId())
+	m.Name = types.StringValue(key.GetLabel())
+	m.PublicKey = types.StringValue(key.GetKey())
+	m.Fingerprint = types.StringValue(key.GetFingerprint())
+	m.Created = types.StringValue(key.CreatedAt.GoString())
+	m.Updated = types.StringValue(key.UpdatedAt.GoString())
+	m.OwnerID = types.StringValue(path.Base(key.GetHref()))
```

Comment:
This is a material change to the logic; the old code was getting the owner ID, but this new code is getting the ID of the SSH key itself.

Reply:
It seems `Owner` field is no longer available in metal-go / API specs but I can see it in the API response. I'm wondering if `Entity` is a new replacement for owner, but still, entity is not in the API response and owner is.

Reply:
It looks like the spec and the API have been mismatched for years on that attribute; the spec should say owner instead of entity.  Options are to (1) open a PR on [equinix-sdk-go](https://github.com/equinix/equinix-sdk-go) that patches the spec to rename entity to owner in the SSHKey schema or (2) use `AdditionalProperties` to work around the spec defect:

```go
// Types are approximate, this might not work
owner = key.AdditionalProperties["owner"].(map[string]interface{})
owner_href = owner["href"].(string)
m.OwnerID = types.StringValue(path.Base(owner_href))
```

Reply:
We're going to need to be careful about accepting minor updates from equinix-sdk-go where new attributes are introduced because of the silent breaking change side-effect of `AdditionalProperties` keys being removed.

To be defensive about that, pushing an error rather than silently failing, we could add more checking:

```go
owner, exists := key.AdditionalProperties["owner"]
if !exists {
    return fmt.Errorf("owner not found")
}

ownerMap, ok := owner.(map[string]interface{})
if !ok {
    return fmt.Errorf("owner is not a map[string]interface{}")
}

ownerHref, exists := ownerMap["href"].(string)
if !exists {
    return fmt.Errorf("href not found or invalid type")
}

m.OwnerID = types.StringValue(path.Base(ownerHref))
```

If getting the ID from an AdditionalProperties map href is common enough, we could make a helper for it.

Reply:
I suspect this would surface in [the same way as it did on the other PR](https://github.com/equinix/terraform-provider-equinix/pull/538#discussion_r1466944138):  an SDK upgrade would cause a panic that would be caught during testing.

IMO patching the SDK is preferable to adding a helper here.  This code is tech debt to work around a short-term issue.  Once we're investing more than a one-off workaround we should do it in a way that benefits all SDK users.